scripts/host_cap.py

```diff
@@ -440,6 +467,117 @@ def add_constituent_vars(cap, host_model, suite_list, run_env):
+                if tend_units.split('s-1')[0].strip() != const_units:
+                    errmsg += f"\nMismatch tendency variable units '{tend_units}'"
+                    errmsg += f" for variable '{tend_stdname}'."
+                    errmsg += f" No variable transforms supported for constituent tendencies."
+                    errmsg += f" Tendency units should be '{const_units} s-1' to match constituent."
+                    prop_error = True
+                # end if
+                # Check the dimensions
+                for index, dim in enumerate(tend_dimensions):
+                    # dimension replacement for horizontal_loop_extent
+                    if dim == 'horizontal_loop_extent':
+                        tend_dim = 'horizontal_dimension'
+                    else:
+                        tend_dim = dim
+                    # end if
+                    if const_dimensions[index] == 'horizontal_loop_extent':
+                        const_dim = 'horizontal_dimension'
+                    else:
+                        const_dim = const_dimensions[index]
+                    # end if
+                    if tend_dim != const_dim:
+                        errmsg += f"\nMismatch tendency variable dimension for dim {index + 1}: '{tend_dim}'"
+                        errmsg += f" for variable '{tend_stdname}'."
+                        errmsg += f" Dimension should match constituent dimension of '{const_dim}'"
+                        prop_error = True
+                    # end if
+                # end for
+                # Check the type
+                if const_type != tend_type:
+                    errmsg += f"\nMismatch tendency variable type '{tend_type}'"
+                    errmsg += f" for variable '{tend_stdname}'."
+                    errmsg += f" Type should match constituent type of '{const_type}'"
+                    prop_error = True
+                # end if
+                # Check the kind
+                if const_kind != tend_kind:
+                    errmsg += f"\nMismatch tendency variable kind '{tend_kind}'"
+                    errmsg += f" for variable '{tend_stdname}'."
+                    errmsg += f" Kind should match constituent kind of '{const_kind}'"
+                    prop_error = True
+                # end if
+                if prop_error == 0:
+                    valid_var = True
+                    exit
+                # end if
+            # end if
+        # end for
+        if not valid_var and not prop_error:
+            # Tendency standard name doesn't match an existing constituent
+            errmsg += f"\nInvalid tendency variable '{tend_stdname}'."
+            errmsg += " All constituent tendency variable standard names must "
+            errmsg += "be of the form 'tendency_of_<constituent_stdname>'"
+            errflg = 1
+        elif prop_error:
+            errflg = 1
+        # end if
+    # end for
+    if errflg != 0:
+        return errmsg, errflg
```

Comment:
```suggestion
```

This `if` is superfluous since the next line does the same thing

Reply:
an excellent point. silly code removed.